### PR TITLE
feat(#major); Belt Finance; Add belt finance

### PR DESCRIPTION
**Pull request by harsh9200**
**Fixes**:
 - Refactored complete source code
 - Update README.
 - Update Revenue logic.
 
 Deployment: [Link](https://thegraph.com/hosted-service/subgraph/harsh9200/belt-finance)

### CODE REVIEW

File: subgraphs/belt-finance/README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Note that Belt applies a Withdrawal and Deposit Fees based on Strategy used at t
 
 Sum across all Vaults:
 
-`(Withdraw Amount * Vault Withdraw Fee) + (Deposit Amount * Vault Deposit Fee)`
+`(Withdraw Amount * Strategy Withdraw Fee) + (Deposit Amount * Strategy Deposit Fee) + Buyback Amount`
```
Comment (this-username-is-taken) [inline subgraphs/belt-finance/README.md:19]: would you update the methodology version here as well as in https://github.com/messari/subgraphs#development-status (should also update subgraph version in the latter)

**Review by this-username-is-taken — APPROVED** [review]
Great work!

File: subgraphs/belt-finance/README.md (hunk 1 above)
Comment (harsh9200) [reply subgraphs/belt-finance/README.md:19]: Done

Comment (this-username-is-taken) [reply subgraphs/belt-finance/README.md:19]: Also don't forget the methodology version

**Pull request merged**